packages/components/src/Bottega/Models/CreditCard.purs

```diff
@@ -0,0 +1,52 @@
+module Bottega.Models.CreditCard where
+
+import Prelude
+
+import Bottega.Models.FailReason (FailReason(..), parseFailReason)
+import Bottega.Models.PaymentMethod (PaymentMethodId)
+import Bottega.Models.PaymentTerminalUrl (PaymentTerminalUrl)
+import Data.Maybe (Maybe, maybe)
+import KSF.Api (UUID)
+
+newtype CreditCardId = CreditCardId Int
+
+type CreditCard =
+  { id              :: CreditCardId
+  , user            :: UUID
+  , paymentMethodId :: Maybe PaymentMethodId
+  , panHash         :: Maybe String
```

Comment:
Actually, can this ever be missing in the UI?

Reply:
Hmm, right, as `POST /paymentMethod/creditCard` creates an empty credit card

Reply:
However from the point of view of just getting the users credit cards, I think the data is invalid if these fields are missing.

Reply:
So we should throw an error?So we should throw an error?

Reply:
So for example when showing the cards to the user, we would need to validate them client side.

We could have a concept of `PendingCreditCard` or `EmptyCreditCard` which would not include these fields. This would also solve the problem of having an optional body in the `POST /paymentMethod/creditCard` endpoint, as then it would always be required. But let's think about this in another time.